pydicom/filewriter.py

```diff
@@ -371,6 +374,9 @@ def write_data_element(fp, data_element, encoding=default_encoding):
     # If explicit VR, write the VR
     VR = data_element.VR
     if not fp.is_implicit_VR:
+        if VR is None:
+            data_element = DataElement_from_raw(data_element, encoding)
+            VR = data_element.VR
```

Comment:
If we got to this stage with explicit VR and the RawDataElement was ambiguous what would happen?

Reply:
Yes, this is the place where the conversion is missing.